### DISCUSSION TITLE
Multiple iperf3 instance causing TCP max throughput to the expected max threshold

### DISCUSSION THREAD

**Discussion by amd-sathekum** [esnet/iperf#1517] [General]
Sometime iperf3 showing results of total accumulated TCP BW is more then the HW line rate.
How is that possible? For example on a 10G NIC card, when we select a speed of **2.5G**. iperf3 showing results as **2.8G.**
we tuned system to max possible

**Comment by davidBar-On** [comment]
Can you give more details about how the NIC card is configured to 2.5G?
It may be that the the 2.5G is "soft" limit, i.e. the card SW tries to make sure that the BW will not be higher.  If this is the case, the the limit cannot be hard limit as with HW limit.  "soft" limit depends for example on the how often its is checked and on the time unit is is calculated.  E.g., is the limit is for every e.g. 0.1 sec separately? Is it averaged over longed time? etc.
